src/hooks/useMedia.ts

```diff
@@ -1,14 +1,18 @@
-import { UseBaseQueryResult, useQuery } from 'react-query';
+import { useQuery } from 'react-query';
 
-import { getMediaById } from '../services/api.service';
+import { getMediaById } from '#src/services/api.service';
+import useSignedUrl from '#src/hooks/useSignedUrl';
 
-import type { PlaylistItem } from '#types/playlist';
+export default function useMedia(mediaId: string, enabled: boolean = true) {
+  const { token, signingEnabled, drmEnabled, drmPolicyId, isLoading } = useSignedUrl('media', mediaId, {}, enabled);
 
-export type UseMediaResult<TData = PlaylistItem, TError = unknown> = UseBaseQueryResult<TData, TError>;
-
-export default function useMedia(mediaId: string, enabled: boolean = true): UseMediaResult {
-  return useQuery(['media', mediaId], () => getMediaById(mediaId), {
-    enabled: !!mediaId && enabled,
+  const queryResult = useQuery(['media', mediaId, token], () => getMediaById(mediaId, token, drmEnabled ? drmPolicyId : undefined), {
```

Comment:
What about putting the `getMediaById` or `getPlaylistById` calls into `useSignedUrl` as callbacks so there isn't as much duplicated code in the boilerplate (the second useQuery) for `useMedia` and `usePlaylist`?

Reply:
Updated this, but realized that this probably wasn't needed because it will be removed in the near future anyway.